staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -125,6 +132,18 @@ func newEnvelopeTransformerWithClock(envelopeService kmsservice.Service, provide
 		providerName:    providerName,
 		stateFunc:       stateFunc,
 		cache:           newSimpleCache(clock, cacheTTL),
+		encryptedObjectDecoder: &encryptedObjectDecoder{
+			bufferPool: &sync.Pool{
+				New: func() interface{} {
+					return proto.NewBuffer(nil)
+				},
+			},
+			objectPool: &sync.Pool{
+				New: func() interface{} {
+					return &kmstypes.EncryptedObject{}
```

Comment:
These objects are a couple bytes. Do we really need to reuse them? The benchmarks for kmsv2 showed that the allocations were dominated everything else in the kube-apiserver. I am skeptical that savings here warrant the additional complexity.